### DISCUSSION TITLE
How to create an ESP-IDF component · Espressif Developer Portal

### DISCUSSION THREAD

**Discussion by giscus** [espressif/developer-portal#346] [General]
# How to create an ESP-IDF component · Espressif Developer Portal

The developer resources in just one place!

https://developer.espressif.com/blog/2024/12/how-to-create-an-esp-idf-component/

**Comment by SkyNetCoder** [comment]
It works for Espressif IDE. But it didn't works for VS Code esp-idf extention: i've got an include errors.

**Reply by kevin-atsuyatech** [reply]
Just add the component name in REQUIRES or PRIV_REQUIRES in the cmakelists.txt inside main/

**Comment by eMUQI** [comment]
> In the next article, we will show how this component can be published to the ESP Component Registry to share with the community.

Hello, I really enjoyed this article. May I ask where I can find the follow-up article about publishing components to the ESP Component Registry, as mentioned at the end?

**Comment by ujur007** [comment]
I have created a component for I2C sensor as well, I would like to contribute. !
